File: Programmers/프로세스.py
def solution(priorities, location):
    answer = 0    
    idx = 0
    while any(priorities):
        if priorities[idx] == max(priorities):
            priorities[idx] = 0
            answer += 1
            if idx == location:
                return answer
        else:
            idx = (idx+1)%len(priorities)

File: Programmers/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int len = priorities.length;
        PriorityQueue<Integer> pq = new PriorityQueue<>((o1,o2)->o2-o1);
        for(int p:priorities) pq.offer(p);
        
        int max;
        int idx = -1;
        for(int cnt = 1;!pq.isEmpty();cnt++){
            max = pq.poll();
 
            for(int i=(idx+1)%len;i!=idx;i=(i+1)%len){
                if(priorities[i]==max){
                    idx = i;
                    break;
                }
            }
            if(idx==location) return cnt;
            
        }
        
        return -1;
    }
}
